modules/op-deploying-api-service-for-tekton-hub.adoc

```diff
@@ -0,0 +1,186 @@
+* For a GitHub OAuth application, set the `Homepage URL` and the `Authorization callback URL` as `<auth-route>`.
+* For a GitLab OAuth application, set the `REDIRECT_URI` as `<auth-route>/auth/gitlab/callback`.
+* For a BitBucket OAuth application, set the `Callback URL` as `<auth-route>`.
+
+. Edit the `02-api/20-api-secret.yaml` file to update the API secret.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: tekton-hub-api
+  namespace: openshift-pipelines <1>
+type: Opaque
+stringData:
+  GH_CLIENT_ID: '' <2>
+  GH_CLIENT_SECRET: '' <3>
+  GHE_URL: '' <4>
+  GL_CLIENT_ID: '' <5>
+  GL_CLIENT_SECRET: '' <6>
+  GLE_URL: '' <7>
+  BB_CLIENT_ID: '' <8>
+  BB_CLIENT_SECRET: '' <9>
+  JWT_SIGNING_KEY: '' <10>
+  ACCESS_JWT_EXPIRES_IN: '' <11>
+  REFRESH_JWT_EXPIRES_IN: '' <12>
+  AUTH_BASE_URL: '' <13>
+----
+<1> The target namespace mentioned in the `TektonHub` CR; default is `openshift-pipelines`.
+<2> The Client ID from the GitHub OAuth application.
+<3> The Client Secret from the GitHub OAuth application.
+<4> GitHub Enterprise URL, if you are authenticating using GitHub Enterprise. Do not provide the URL to the catalog as a value for this field.
+<5> The Client ID from the GitLab OAuth application.
+<6> The Client Secret from the GitLab OAuth application.
+<7> GitLab Enterprise URL, if you are authenticating using GitLab Enterprise. Do not provide the URL to the catalog as a value for this field.
+<8> The Client ID from the BitBucket OAuth application.
+<9> The Client Secret from the BitBucket OAuth application.
+<10> A long, random string used to sign the JSON Web Token (JWT) created for users.
+<11> Add the time limit after which the access token expires. For example, `1m`, where `m` denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`).
+<12> Add the time limit after which the refresh token expires. For example, `1m`, where `m` denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`). Ensure that the expiry time set for token refresh is greater than the expiry time set for token access.
+<13> Route URL for the OAuth application.
++
+[NOTE]
+====
+Use the fields related to Client ID and Client Secret for any one of the supported Git repository hosting service providers.
+====
+
+. Update the API config map.
+.. Update the `config.yaml` file to include at least one user with the following permissions:
+* refresh a catalog
+* refresh a config file
+* create an agent token. 
++
+[source,yaml]
+----
+...
+scopes:
+- name: agent:create
+  users: <your_GitHub_username> <1>
+- name: catalog:refresh
+  users: <your_GitHub_username> <2>
+- name: config:refresh
+  users: <your_GitHub_username> <3>
+...
+----
+<1> An user with `agent:create` scope can set up a cron job that refreshes the {tekton-hub} database after an interval, if there are any changes in the catalog.
```

Comment:
```suggestion
<1> A user with `agent:create` scope can set up a cron job that refreshes the {tekton-hub} database after an interval, if there are any changes in the catalog.
```
Fix seven instances of "an user" in this PR.

Reply:
IBM SG: Choose the indefinite article a or an according to how an abbreviation [or word] is spoken. Use a before an
abbreviation that begins with a consonant sound. Use an before an abbreviation with a vowel sound.

Reply:
If you adopt this suggestion, please update other similar instances in this PR.